src/agimus_pytroller.cpp

```diff
@@ -217,6 +217,55 @@ controller_interface::CallbackReturn AgimusPytroller::on_configure(
   return controller_interface::CallbackReturn::SUCCESS;
 }
 
+inline std::vector<char> AgimusPytroller::msg_to_buffer(
+    const std::shared_ptr<rclcpp::SerializedMessage> &msg) {
+  const auto serialized_msg = msg->get_rcl_serialized_message();
+  std::vector<char> payload(serialized_msg.buffer,
+                            serialized_msg.buffer +
+                                serialized_msg.buffer_length);
+  return payload;
+}
+
+std::shared_ptr<rclcpp::SerializedMessage> AgimusPytroller::fetch_message_once(
+    const std::shared_ptr<rclcpp::Node> &temp_node,
+    const std::string &subscriber_name) {
+
+  const auto &topic_params =
+      params_.initialization_data_topics_map.at(subscriber_name);
+
+  const auto &topic_type = topic_params.topic_type;
+  const auto topic_name = topic_params.topic_name;
+
+  std::shared_ptr<rclcpp::SerializedMessage> topic_data;
+  const auto qos = topic_params.transient_local
+                       ? rclcpp::QoS(1).transient_local()
+                       : rclcpp::SystemDefaultsQoS();
+  auto topic_sub = temp_node->create_generic_subscription(
+      topic_name, topic_type, qos,
+      [&](const std::shared_ptr<rclcpp::SerializedMessage> msg) {
+        topic_data = msg;
+      });
+
+  for (int i = 0; i < topic_params.retires; i++) {
```

Comment:
```suggestion
  for (int i = 0; i < topic_params.retries; i++) {
```
and all the others ;)

Reply:
Fixed